clang/test/CodeGenSYCL/scheduler-target-fmax-mhz.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -disable-llvm-passes -triple spir64-unknown-unknown-sycldevice -emit-llvm -o - %s | FileCheck %s
+
+#include "Inputs/sycl.hpp"
+[[intelfpga::scheduler_target_fmax_mhz(5)]] void
+func() {}
+
+template <int N>
+[[intelfpga::scheduler_target_fmax_mhz(N)]] void zoo() {}
+
+int main() {
+  cl::sycl::kernel_single_task<class test_kernel1>(
+      []() [[intelfpga::scheduler_target_fmax_mhz(2)]]{});
+
+  cl::sycl::kernel_single_task<class test_kernel2>(
+      []() { func(); });
+
+  cl::sycl::kernel_single_task<class test_kernel3>(
+      []() { zoo<75>(); });
+}
+// CHECK: define spir_kernel void @{{.*}}test_kernel1() {{.*}} !scheduler_target_fmax_mhz ![[PARAM1:[0-9]+]]
+// CHECK: define spir_kernel void @{{.*}}test_kernel2() {{.*}} !scheduler_target_fmax_mhz ![[PARAM2:[0-9]+]]
```

Comment:
So the attribute is automatically promoted up to he kernel through the mechanism implemented in SemaSYCL, right?

Reply:
Correct. I added this attribute to "CollectPossibleKernelAttributes" and "MarkDevice" in SemaSYCL to do just that.